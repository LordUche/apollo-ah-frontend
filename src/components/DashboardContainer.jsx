import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { getDashboard, getDashboardSummary } from '../selectors/dashboardSelector';
import { getProfile } from '../selectors/profileSelector';
import Dashboard from '../views/Dashboard';
import ProfileHeaderContainer from './ProfileHeaderContainer';
import { getDashboardArticles } from '../actions/dashboardAction';
import CommentLoader from '../views/CommentLoader';

/**
 * Container component for the Dashboard view
 * @export
 * @class DashboardContainer
 * @extends {Component}
 */
class DashboardContainer extends Component {
  async componentDidMount() {
    const { profile: { id } } = this.props;
    const { getDashboardArticles: fetchDashboardArticles } = this.props;
    await fetchDashboardArticles(id);
  }

  render() {
    const { dashboard, summary } = this.props;
    return (
      <>
        <ProfileHeaderContainer activePage="DASHBOARD" />
        {
          dashboard.loading
            ? <CommentLoader />
            : <Dashboard summary={summary} dashboard={dashboard} />
        }
      </>
    );
  }
}

DashboardContainer.propTypes = {
  dashboard: PropTypes.object.isRequired,
  summary: PropTypes.object.isRequired,
  profile: PropTypes.object.isRequired,
  getDashboardArticles: PropTypes.func.isRequired,
};

const mapStateToProps = createStructuredSelector(
  {
    dashboard: getDashboard,
    summary: getDashboardSummary,
    profile: getProfile,
  }
);
const mapDispatchToProps = dispatch => ({
  getDashboardArticles: payload => dispatch(getDashboardArticles(payload)),
});
export default connect(mapStateToProps, mapDispatchToProps)(DashboardContainer);
