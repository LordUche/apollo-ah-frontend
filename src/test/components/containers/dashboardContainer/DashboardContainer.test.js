import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import DashboardContainer from '../../../../components/containers/DashboardContainer';

configure({ adapter: new Adapter() });

describe('<DashboardContainer>', () => {
  it('should render', () => {
    const wrapper = shallow(<DashboardContainer />);
    expect(wrapper).toMatchSnapshot();
  });
});
