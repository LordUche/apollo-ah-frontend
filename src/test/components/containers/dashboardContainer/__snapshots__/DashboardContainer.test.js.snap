// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DashboardContainer> should render 1`] = `
<Fragment>
  <Connect(ProfileHeaderContainer)
    activePage="DASHBOARD"
  />
  <Dashboard />
</Fragment>
`;
